GomokuClient/packages/gomoku-core/src/hooks/useSignalRReconnection.ts

```diff
@@ -1,34 +1,43 @@
 import { useEffect } from "react";
 import * as signalR from "@microsoft/signalr";
+import { CellValue } from "@/hooks/useBoard";
 
 interface PlayerJoinedGameServerMessage {
-  gameId: string;
+  userName: string;
 }
 
 interface PlayerMadeMoveServerMessage {
   playerId: string;
   tile: TileItem;
+  placedTileColor: CellValue;
 }
 
 interface TileItem {
   x: number;
   y: number;
 }
 
+interface GameHubError {
+  message: string;
+}
+
 interface UseSignalRReconnectionProps {
   connection: signalR.HubConnection | null;
   isConnected: boolean;
   onPlayerJoined?: (message: PlayerJoinedGameServerMessage) => void;
   onPlayerMadeMove?: (message: PlayerMadeMoveServerMessage) => void;
   onReceiveMessage?: (user: string, message: string) => void;
+  onGameHubError?: (error: GameHubError) => void;
 }
 
+// TODO: Is it good name for this hook? Maybe useGameHub fits more to what it really does?
```

Comment:
- Originally, this was my quick test to challenge the reconnection state of a more stable connection, ensuring that listeners could be reattached if they were disconnected or if an error occurred.

- Ideally, this hook shouldn’t exist, as the reconnection logic should be encapsulated directly within the SignalRContext.

- The reasoning behind the current name is that the hook’s primary role is to attach listeners and manage the lifecycle of a SignalR connection

Reply:
Okay, got it. Will remove comment